src/sleep/sleep.c

```diff
@@ -274,7 +274,7 @@ static int custom_timer_suspend(const SleepConfig *sleep_config) {
 
         hibernate_timestamp = usec_add(now(CLOCK_BOOTTIME), sleep_config->hibernate_delay_usec);
 
-        while (battery_is_low() == 0) {
+        while (!battery_is_present_and_low()) {
```

Comment:
Please don't use `!` on a value which may be < 0. The previous version with `== 0` was better.